### PR TITLE
Adding Swift templates to Meson

### PR DESCRIPTION
**Pull request by michaelbadcrumble**
Finally found some time to implemented Swift programming language templates. I also added a note and updated the test case to find the swift compiler.

### CODE REVIEW

**Comment by michaelbadcrumble** [comment]
Ping @dcbaker and or @jpakkane.

File: run_unittests.py [hunk 1]
```diff
@@ -3404,6 +3404,12 @@ def test_templates(self):
                 langs.append('rust')
             except EnvironmentException:
                 pass
+        if is_osx():
```
Comment (dcbaker) [inline run_unittests.py:3407]: Everything looks good here except this, swift works on oses other than macos.

Comment (michaelbadcrumble) [reply run_unittests.py:3407]: So. Dose it work on Linux?

Comment (michaelbadcrumble) [reply run_unittests.py]: Ok did some digging and turns out Swift can run on Linux.

Comment (michaelbadcrumble) [reply run_unittests.py]: It’s not looking good doc. What do I do from here @dcbaker ?

**Comment by dcbaker** [comment]
one of the documented commands tests is failing, you should be able to fix that pretty easily

**Comment by michaelbadcrumble** [comment]
I have opened a new PR for swift template support with the latest frock to the Meson build project and was finally able to find that the documented commands test case. Should be here (#7339).

Also, there is an issue with the Swift support please go see the issue (#7337), thanks.

**Pull request closed without merging**